### ISSUE TITLE
Positive/Negative Insets

### ISSUE THREAD

**Issue by laurmaedje** [linebender/kurbo#140]
Positive `Insets` in `kurbo` make a rectangle larger and negative ones smaller. I find this pretty confusing because I think of _insetting_ something as moving it _inwards_. From a very quick search, this is also how it's done in [UIKit], [Android Graphics] and in [Java AWT]. Maybe there's maybe a good reason why it's different here (and it's also a bit bug-prone change to do now) but I still wanted to point it out because I think its kind of confusing.

[UIKit]: https://developer.apple.com/documentation/uikit/uiedgeinsets
[Android Graphics]: https://developer.android.com/reference/android/graphics/Insets
[Java AWT]: https://docs.oracle.com/javase/7/docs/api/java/awt/Insets.html

**Comment by derekdreery** [comment]
According to [the dictionary definition](https://dictionary.cambridge.org/dictionary/english/inset), an inset is something larger that something smaller is set in. I would interpret positive values as the "default" - so larger. This also matches the intuition that positive means bigger, while negative means smaller. All of this isn't fact, just my 2 cents.

**Comment by cmyr** [comment]
I agree this is confusing, and I'm not sure how best to think about it.

The rationale for the current implementation is that we implement both addition and subtraction of insets, and I think it is most intuitive if (given positive insets) `(rect + insets).area() > (rect - insets).area()`.

It's very possible I got this backward, I remember feeling confused at the time.

**Comment by laurmaedje** [comment]
I see where you're coming from with the addition/subtraction thing. 

I think, the main thing is that in the current implementation addition is _the_ way to "apply" insets to a rectangle. Maybe it's equally possible to see subtraction as the default way of "applying" insets. Then, positive insets shrink a rect and we have 
```rust
rect - positive_insets => smaller_rect
rect + positive_insets => larger_rect
```

I'm not sure, but it might even be okay to not actually specify whether positive insets shrink or expand a rectangle. Instead, just define the plus and minus operations and then let its usage define what an `Inset` means. Here's an example:
```rust
let margins = Insets::uniform(5.0);
let padding = Insets::uniform(10.0);

let smaller_rect = rect - padding;
let larger_rect = rect + margins;
```
This is kind of like in CSS where even though padding goes inwards and margins outwards, both are assigned positive values. It's possible that this is a really bad idea though, it's just a thought.

**Comment by derekdreery** [comment]
I really don't know what the right answer is here. I like the simplicity of the current setup, but agree it could be confusing. Perhaps this is another documentation question - perhaps we could make the `Inset` docs better?

**Comment by cmyr** [comment]
I'm happy with our choice, and I think that the [docs for `Insets`](https://docs.rs/kurbo/latest/kurbo/struct.Insets.html) are pretty good, so I'm inclined to close this.
